File: csharp/ClipboardLogger.cs
using System;
using System.IO;
using System.Threading;
using System.Windows.Forms;

class Program
{
    [STAThread]
    static void Main()
    {
        string ultimo = "";

        Console.WriteLine("Monitorando Ctrl+C...");

        while (true)
        {
            if (Clipboard.ContainsText())
            {
                string texto = Clipboard.GetText();

                if (texto != ultimo)
                {
                    File.AppendAllText(
                        "CsharpClipboard.txt",
                        texto + Environment.NewLine
                    );
                    ultimo = texto;
                }
            }

            Thread.Sleep(1000);
        }
    }
}

File: java/ClipboardLogger.java
import java.awt.Toolkit;
import java.awt.datatransfer.DataFlavor;
import java.io.FileWriter;

public class ClipboardLogger {
    public static void main(String[] args) throws Exception {
        String ultimo = "";

        System.out.println("Monitorando Ctrl+C...");

        while (true) {
            String texto = (String) Toolkit.getDefaultToolkit()
                    .getSystemClipboard()
                    .getData(DataFlavor.stringFlavor);

            if (!texto.equals(ultimo)) {
                FileWriter fw = new FileWriter("Clipboard_Logger/java/JavaClipboard.txt", true);
                fw.write(texto + "\n");
                fw.close();
                ultimo = texto;
                System.out.println("Texto salvo");
            }
            Thread.sleep(1000);
        }
    }
}